specifyweb/specify/build_models.py

```diff
@@ -112,11 +96,13 @@ def make_relationship(modelname, rel, datamodel):
         return None
 
     try:
-        on_delete = SPECIAL_DELETION_RULES["%s.%s" % (modelname.capitalize(), rel.name.lower())]
+        on_delete = SPECIAL_DELETION_RULES[f"{rel.name.capitalize()}"][f"{modelname.lower()}"]
```

Comment:
It looks like you swapped the relationship and model name. Is this intentional?

Reply:
I think I should change the variable names...

This was confusing even in production. 

Take this for example. 
https://github.com/specify/specify7/blob/741a5e34da50328ce8706d25c2071faba3976a2e/specifyweb/specify/build_models.py#L77
This operates as 'When a Specifyuser is deleted, set its relationship with Agent to null"
Where the on_delete functionality for these rules get defined, the `modelname` would be Agent, and `rel.name` would be `specifyuser`.
https://github.com/specify/specify7/blob/741a5e34da50328ce8706d25c2071faba3976a2e/specifyweb/specify/build_models.py#L115

I thought it might make more sense to have the table being deleted be the 'primary' key of the dictonary, so I changed it to `'Specifyuser' : {'agent': models.SET_NULL}`

As a result of this flip, I flipped the variables where `on_delete` was being defined to ensure the same functionality.

Reply:
I can change it to `'Agent': {'specifyuser': models.SET_NULL}` if that makes more sense

Reply:
I think adding an example breakdown you provided here as a comment on top of `SPECIAL_DELETION_RULES` (in addition to what you already have there) would be great